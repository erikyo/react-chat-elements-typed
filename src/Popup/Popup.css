.rce-popup-wrapper {
    position: fixed;
    inset: 0;
    margin: auto;
    z-index: 999999;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
}

.rce-popup {
    background: var(--rce-color-white);
    border-radius: var(--rce-mbox-border-radius);
    padding: 0 10px;
    width: 400px;
    min-height: 100px;
    box-shadow: 0 5px 20px -10px var(--rce-color-black);
    border: 1px solid var(--rce-color-gray-100);
    display: flex;
    flex-direction: column;
    align-items: stretch;
    justify-content: flex-start;
    animation: popup-scaling 0.4s ease forwards;
    box-sizing: border-box;
}

@keyframes popup-scaling {
    0% {
        transform: scale(0);
        opacity: 0;
    }

    50% {
        transform: scale(1.2);
        opacity: 0.5;
    }

    100% {
        transform: scale(1);
        opacity: 1;
    }
}

.rce-popup-header {
    padding: 18px 8px 0;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    box-sizing: border-box;
    font-weight: 800;
    text-transform: capitalize;
}

.rce-popup-header .rce-button {
    padding: 0;
}


.rce-popup-content {
    padding: 8px;
    font-size: 14px;
    box-sizing: border-box;
}

.rce-popup-content * {
    margin: 0;
}

.rce-popup-footer {
    padding: 18px 8px;
    display: flex;
    flex-direction: row;
    align-items: center;
    gap: 0.5rem;
    justify-content: flex-end;
    box-sizing: border-box;
}
.rce-popup-footer button {
    min-width: 100px;
}
