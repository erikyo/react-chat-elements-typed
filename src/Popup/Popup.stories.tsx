import type { StoryObj } from "@storybook/react";
import React, { useState } from "react";
import Popup from "./Popup";
import Button from "../Button/Button";
import type { IPopup } from "../types";
import { MdClose } from "react-icons/md";
import { loremIpsum } from "lorem-ipsum";

const Template = {
	header: loremIpsum({ count: 2, units: "words" }),
	text: loremIpsum({ count: 4, units: "sentence" }),
	wrapperColor: "rgb(0 0 0 / 75%)",
} as IPopup;

const PopupExample = () => {
	const [show, setShow] = useState<boolean>(false);
	return (
		<div style={{ margin: "45vh" }}>
			<Popup
				popup={{
					...Template,
					headerButtons: [
						{
							color: "black",
							backgroundColor: "transparent",
							onClick: () => {
								setShow(false);
							},
							icon: {
								backgroundColor: "transparent",
								component: <MdClose color={"var(--rce-color-gray)"} />,
								size: 24,
							},
						},
					],
					footerButtons: [
						{
							color: "white",
							backgroundColor: "var(--rce-color-gray)",
							text: "No",
							squared: true,
							outlined: true,
							onClick: () => {
								setShow(false);
							},
						},
						{
							color: "white",
							backgroundColor: "var(--rce-color-primary)",
							text: "Yes",
							squared: true,
							onClick: () => {
								setShow(false);
							},
						},
					],
				}}
				show={show}
			/>
			<Button
				text="Show Popup"
				onClick={() => {
					setShow(true);
				}}
			/>
		</div>
	);
};

const meta = {
	component: PopupExample,
	parameters: {
		layout: "fullscreen",
	},
};

export default meta;

type Story = StoryObj<typeof meta>;

export const Default = {
	args: { popup: Template },
};
