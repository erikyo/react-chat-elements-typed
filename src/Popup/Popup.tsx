import type React from "react";
import type { FC } from "react";
import "./Popup.css";

import Button from "../Button/Button.js";

import classNames from "classnames";
import type { IPopupProps } from "../type.js";

const Popup: FC<IPopupProps> = ({ ...props }) => {
	if (props.popup?.show === true)
		return (
			<div
				className={classNames("rce-popup-wrapper", props.type, props.className)}
			>
				<div className="rce-popup">
					{props.popup?.renderHeader ? (
						<div className="rce-popup-header">
							{props.popup?.renderHeader()}
						</div>
					) : (
						<div className="rce-popup-header">
							<span>{props.popup?.header}</span>
							{props.popup?.header &&
								props.popup?.headerButtons?.map((x, i) => (
									<Button key={i.toString()} {...x} />
								))}
						</div>
					)}
					<div
						className="rce-popup-content"
						style={{ color: props.popup?.color }}
					>
						{props.popup?.renderContent
							? props.popup?.renderContent()
							: props.popup?.text}
					</div>
					<div className="rce-popup-footer">
						{props.popup?.renderFooter
							? props.popup?.renderFooter()
							: props.popup?.footerButtons?.map((x, i) => (
									<Button key={i.toString()} {...x} />
								))}
					</div>
				</div>
			</div>
		);
	return null;
};

export default Popup;
