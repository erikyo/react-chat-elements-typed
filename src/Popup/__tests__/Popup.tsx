import { describe, expect, it } from "vitest";
import React from "react";
import Popup from "../Popup";
import { render } from "@testing-library/react";

describe("Avatar component", () => {
	it("should render without issues", () => {
		const component = render(<Popup show={true} />);
		expect(component.container).toBeTruthy();
		expect(component.container).toMatchSnapshot();
	});
});
