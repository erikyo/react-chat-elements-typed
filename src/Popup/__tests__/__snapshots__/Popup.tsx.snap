// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar component > should render without issues 1`] = `
<div>
  <div
    class="rce-popup-wrapper"
  >
    <div
      class="rce-popup"
    >
      <div
        class="rce-popup-header text-extrabold text-2xl"
      >
        <span />
      </div>
      <div
        class="rce-popup-content"
      />
      <div
        class="rce-popup-footer"
      />
    </div>
  </div>
</div>
`;
